app/controllers/api/v1/application_controller.rb

```diff
@@ -0,0 +1,8 @@
+class Api::V1::ApplicationController < ActionController::Base
+  include CurrentUser
+  
+  def login_user
+    warden.authenticate(:basic) || unauthorized!
```

Comment:
Not sure I understand...
Do you mean to define a way for a class that includes the `CurrentUser` concern to override the "allowed auth methods"? so there is no need to override `login_user`?

What do you think about:
```ruby
class Api::V1::ApplicationController < ActionController::Base
  include CurrentUser

  def warden_auth_strategies
    [:basic]
  end
end
```

where the default implementation of `warden_auth_strategies` will return an empty array (which means the value of `default_strategies` as it was set in `config/initializers/warden.rb` will be used)